src/components/menu-surface/menu-surface.tsx

```diff
@@ -106,12 +112,32 @@ export class MenuSurface {
     }
 
     private handleDocumentClick(event) {
-        if (this.open && !isDescendant(event.target, this.host)) {
+        const elementPath = event.path || [];
+
+        const clickedInAllowedElement = this.isElementInPath(
+            elementPath,
+            this.allowClicksElement
```

Comment:
Awesome! This was the once thing I had a comment about when I looked it over earlier, because a click in any _other_ picker would have also been allowed to propagate, and wouldn't close the picker 👍

Reply:
Tiny question though: Rather than looking at the path, can't we use `isDescendant` for this check as well?

Reply:
```tsx
isDescendant(event.target, this.allowClicksElement);
```

Reply:
Then we can't know which element that was clicked since the event.target is the document?

Reply:
Hmm… I'm pretty sure event.target should still be the element that was clicked, not the element on which the handler was registered…? Are you sure? 🤔

Reply:
Yes since it's shadow doms! :/

Reply:
Whaa… really? Now I'm confused and curious! 😄

Imma fiddle around a bit, and see if I can understand what's happening… 🤔

Reply:
Wow, yeah, you're absolutely right 👍

Good thing I found out about that little quirk before it bit me! 🙏